src/tree/subscriptionTreeItem.ts

```diff
@@ -73,43 +82,115 @@ class SubscriptionTreeItem extends AzExtParentTreeItem implements SubscriptionTr
+        const fleetToMembersMap = new Map<string, DefinedFleetMemberWithGroup[]>();
+        const clusterToMemberMap = new Map<string, DefinedFleetMemberWithGroup>();
+
+        const memberPromises = fleetResources.map(async (f) => {
+            const members = await getFleetMembers(this.sessionProvider, f);
+            if (failed(members)) {
+                window.showErrorMessage(
+                    `Failed to list fleets in subscription ${this.subscriptionId}: ${members.error}`,
+                );
+                return null;
+            }
+            // filter out members that do not satisfy the filter (so that they are not shown in the tree)
+            const membersAfterFilt = members.result.filter((m) => {
+                // for each member cluster of the fleet
+                const filteredClusters = getFilteredClusters();
+                return filteredClusters.some(
+                    // check if the member is one of the clusters in the filter
+                    (filter) =>
+                        filter.subscriptionId === this.subscriptionId &&
+                        filter.clusterName === parseResource(m.clusterResourceId).name,
+                );
+            });
+            // key - fleet.id, val: a list of all members of the fleet except the ones that do not satisfy the filter
+            fleetToMembersMap.set(f.id, membersAfterFilt);
+            return membersAfterFilt;
+        });
+        await Promise.all(memberPromises); // wait for all members to be fetched
+
+        fleetToMembersMap.forEach((members) => {
+            members.forEach((member) => {
+                clusterToMemberMap.set(member.clusterResourceId.toLowerCase(), member);
+            });
+        });
+
+        return { fleetToMembersMap, clusterToMemberMap };
+    }
+
+    public async loadMoreChildrenImpl(): Promise<AzExtTreeItem[]> {
+        let clusterResources: DefinedResourceWithGroup[] = [];
+        let fleetResources: DefinedResourceWithGroup[] = [];
+        ({ clusterResources, fleetResources } = await this.fetchClustersAndFleets());
+        const { fleetToMembersMap, clusterToMemberMap } = await this.mapFleetAndClusterMembers(fleetResources);
+
+        // remove clusters that are members of fleets
+        clusterResources = clusterResources.filter((r) => !clusterToMemberMap.has(r.id.toLowerCase()));
+
+        // create tree nodes for filtered clusters and fleets
+        const filteredClusters = getFilteredClusters();
+
+        const fleetTreeNodes = new Map<string, FleetTreeNode>();
+        const clusterTreeItems = new Map<string, AzExtTreeItem>();
+        fleetResources.concat(clusterResources).forEach((r) => {
+            if (r.type?.toLocaleLowerCase() === fleetResourceType.toLowerCase()) {
+                const fleetTreeItem = createFleetTreeNode(this, this.subscriptionId, r);
+                fleetTreeItem.addCluster(fleetToMembersMap.get(r.id) || []);
```

Comment:
I have tested this using the VSIX  but for some of the subscriptions it doesn't show me the member clusters under the fleet tree view, which could be this line causing it or some role assignments are missing. Comparing it with portal, portal does few more steps which might be missing here. But that can be done as incremental PR. Good Job on this PR though @JunyuQian 
![image](https://github.com/user-attachments/assets/af434e0d-b6d5-4803-b277-cb198164f2a4)